### ISSUE TITLE
device-simple crash on startup

### ISSUE THREAD

**Issue by tobiasmo1** [edgexfoundry/device-sdk-go#230]
The device-simple exits with a segmentation fault after attempting to load configuration.toml.

**Scenario 1 - Running from GoLand/other IDE:**
The default working directory that populates a developer's runtime configuration after cloning and loading DSDK project will be:
> /home/_user_/go/src/github.com/edgexfoundry/device-sdk-go

Note that it will not implicitly append `/example/cmd/device-simple/`

**Scenario 2 - Running from terminal:**
When running device-simple from an external folder the same occurs.
1. set working directory (e.g., base project folder)
`cd /home/_user_/go/src/github.com/edgexfoundry/device-sdk-go`
2.  launch the built application
`./example/cmd/device-simple/device-simple`

In either scenario the configuration.toml fails to load, resulting in the segfault due to dereference of config variable.

Remedy, report working directory to help developer identify the issue. Also check for error when configuration is not successfully loaded.

**Comment by tsconn23** [comment]
Be advised there's a cmd line parameter allowing a user to specify where the config directory is in cases where the runtime location is non-standard. Example, I have to use this for everything I run in GoLand.

```
flag.StringVar(&confDir, "confdir", "", "Specify an alternate configuration directory.")
flag.StringVar(&confDir, "c", "", "Specify an alternate configuration directory.")
```

**Comment by tobiasmo1** [comment]
Understood. If a full path is specified supplying the confdir works well. The default is using a relative path (./res).
As Trevor calls out, the implementation must also honor the alternate configuration directory (if specified).

**Comment by tobiasmo1** [comment]
+ @SteveOss  
The corresponding behavior expected from the profile cmdline parameter is unclear to me however. Is this always expected to be a full path to a device profile .yaml file, or an alternate name of a device profile .yaml file found within the confDir?

```
flag.StringVar(&confProfile, "profile", "", "Specify a profile other than default.")
flag.StringVar(&confProfile, "p", "", "Specify a profile other than default.")

```

**Comment by tobiasmo1** [comment]
To be more specific, I see entries within the configuration.toml that declare profile location (i.e., distinct from confDir).

https://github.com/edgexfoundry/device-sdk-go/blob/master/example/cmd/device-simple/res/configuration.toml#L52


I also see distinct profile assigned to each device being created. A single profile parameter would not seem to support this configuration .toml paradigm.
https://github.com/edgexfoundry/device-sdk-go/blob/master/example/cmd/device-simple/res/configuration.toml#L61

**Comment by cloudxxx8** [comment]
fixed by #231
